be/src/exec/join_hash_map.tpp

```diff
@@ -820,20 +826,28 @@ void JoinHashMap<LT, BuildFunc, ProbeFunc>::_search_ht_remain(RuntimeState* stat
     _probe_state->count = match_count;
 }
 
-#define DO_PROBE(X, Y)                                                                  \
-    if (_probe_state->active_coroutines != 0) {                                         \
-        if constexpr (first_probe) {                                                    \
-            auto group_size = std::abs(state->query_options().interleaving_group_size); \
-            _probe_state->cur_probe_index = 0;                                          \
-            _probe_state->handles.clear();                                              \
-            for (int i = 0; i < group_size; ++i) {                                      \
-                _probe_state->handles.insert(X(state, build_data, data));               \
-            }                                                                           \
-            _probe_state->active_coroutines = group_size;                               \
-        }                                                                               \
-        _probe_coroutine<first_probe, Y>(state, build_data, data);                      \
-    } else {                                                                            \
-        X<first_probe>(state, build_data, data);                                        \
+#define DO_PROBE(X, Y)                                                                                          \
+    if (_probe_state->active_coroutines != 0) {                                                                 \
+        if constexpr (first_probe) {                                                                            \
+            auto group_size = std::abs(state->query_options().interleaving_group_size);                         \
+            _probe_state->cur_probe_index = 0;                                                                  \
+            if (!_probe_state->handles.empty()) {                                                               \
+                for (auto& h : _probe_state->handles) {                                                         \
+                    h.destroy();                                                                                \
+                }                                                                                               \
+                _probe_state->handles.clear();                                                                  \
+                std::string msg =                                                                               \
+                        "fragment = " + print_id(state->fragment_instance_id()) + " 's probe leaks coroutines"; \
+                throw std::runtime_error(msg + ", please set interleaving_group_size = 0 to disable it");       \
```

Comment:
I think this error message is not friendly, and users may be confused after seeing it. can we find problems during testing? e.g. add DCHECK here

Reply:
IF really occur again, the log may be ignored, so here should throw exceptions. I will refine the log.